common4j/src/main/com/microsoft/identity/common/java/challengehandlers/PKeyAuthChallenge.java

```diff
@@ -179,12 +181,17 @@ public String getSubmitUrl() {
                 if (privateKey == null) {
                     throw new ClientException(ErrorStrings.KEY_CHAIN_PRIVATE_KEY_EXCEPTION);
                 }
+                final PublicKey publicKey = deviceCertProxy.getPublicKey();
+                final X509Certificate certificate = deviceCertProxy.getCertificate();
+                if (certificate == null) {
```

Comment:
What if the device isn't joined? Or we are not going to even get here in that case?

Reply:
the cert proxy should not be set in the first place